lib/resources/registry_key.rb

```diff
@@ -66,7 +66,10 @@ def initialize(name, reg_key = nil)
         # generate registry_key if we do not have a regular expression
         @options[:path] = @options[:hive]
         # add optional key path
-        @options[:path] += '\\' + @options[:key] if @options[:key]
+        if @options[:key]
```

Comment:
Can we break out the building of the path to a private helper method in this class and add some tests to it? There's enough conditional logic in building the path now that I think it's warranted.